kotlin-checks-test-sources/src/main/kotlin/checks/CoroutinesTimeoutApiUnusedCheckSample.kt

```diff
@@ -0,0 +1,101 @@
+package checks
+
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.async
+import kotlinx.coroutines.coroutineScope
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withTimeout
+import kotlinx.coroutines.withTimeoutOrNull
+
+class CoroutinesTimeoutApiUnusedCheckSample {
+    suspend fun foo() {
+        coroutineScope {
+            println("")
+            // Noncompliant@+8
+            val job = launch {
+//                    ^^^^^^>
+                delay(2000L)
+                println("Finished")
+            }
+            delay(500L)
+//          ^^^^^>
+            job.cancel()
+//              ^^^^^^
+        }
+
+        coroutineScope {
+            println("")
+            val job = async() {
```

Comment:
Looks like it's better to name the result `deferred` and add examples where result of `async` function is actually used (with relevant compliant example) to make sure we're not suggesting anything invalid.

For example:

```
        coroutineScope {
            val deferred = async() {
                delay(2000L)
                println("Finished")
                100
            }
            delay(500L)
            deferred.cancel() // Noncompliant
            deferred.await()
        }

        coroutineScope {
            println("")
            val i = withTimeoutOrNull(500L) {
                delay(2000L)
                println("Finished")
                100
            }
            i
        }
```